### PR TITLE
[cmake][llvm] Limit the number of Xcode schemes created by default

### PR DESCRIPTION
**Pull request by jroelofs**
CMake -GXcode would otherwise offer to create one scheme for each target, which ends up being a lot. For now, limit the default to the `check-*` LIT targets, plus `ALL_BUILD` and `install`.

For targets that aren't in the default list, we now have a configuration variable to promote an extra list of targets into schemes, for example `-DLLVM_XCODE_EXTRA_TARGET_SCHEMES="TargetParserTests;SupportTests"` to add schemes for `TargetParserTests` and `SupportTests` respectively.

### CODE REVIEW

**Comment by JDevlieghere** [comment]
Nice. An added benefit of doing it this way is that you can always set the property for a particular target if you really need the scheme and you can do it in the same cmake file that creates the target (i.e. in a subproject).

**Comment by jroelofs** [comment]
And you can always manually override them through the scheme editor thing, if that is a one-off local need.

**Comment by clayborg** [comment]
Can we add a cmake settings that allows users to add custom target names that will be added? I would love to use this for my common projects like:
```
-DCMAKE_LLVM_XCODE_TARGETS=lldb;llvm-dwarfdump;llvm-gsymutil;lld
```
And it would add those top level targets?

**Comment by jroelofs** [comment]
neat idea. that's doable.

**Review by JDevlieghere — APPROVED** [review]
LGTM!

**Comment by clayborg** [comment]
Great! That will be very useful.

File: llvm/CMakeLists.txt [hunk 1]
```diff
@@ -1423,3 +1423,11 @@ endif()
 if (LLVM_INCLUDE_UTILS AND LLVM_INCLUDE_TOOLS)
   add_subdirectory(utils/llvm-locstats)
 endif()
+
+if (XCODE)
+  set(LLVM_XCODE_EXTRA_TARGET_SCHEMES "" CACHE STRING "Specifies an extra list of targets to turn into schemes")
```
Comment (clayborg) [inline llvm/CMakeLists.txt:1428]: Do we want to say "an extra list of semi colon separated target names"? Do you specify this multiple times, or just once with semi colon separated target names?

Comment (jroelofs) [reply llvm/CMakeLists.txt:1428]: Lists in CMake are semicolon-separated. I'll add an example in a comment.

**Pull request merged**